### PR TITLE
LAB-253: Refactor unit tests

### PR DESCRIPTION
**Pull request by gita135**
Refactored unit tests for type refactor after testing on all instruments `TcpInstrurments.jl` supports. Also fixed typos from type refactor

### CODE REVIEW

File: test/pwr/bench_xr.jl [hunk 1]
```diff
@@ -31,53 +37,41 @@ end
     function test_current(cur)
         set_current_limit(pwr, cur)
         @info get_current_limit(pwr)
-        @test get_current_limit(pwr) == cur
+        @test uparse(get_current_limit(pwr)) * A== cur
```
Comment (deepaksnn) [inline test/pwr/bench_xr.jl:40]: space before "=="

File: test/pwr/agilent_e36312a.jl [hunk 2]
```diff
@@ -76,19 +79,15 @@ end
...
-        test_volt(4, i)
-        test_volt(6, i)
+        test_volt(2 * V, i)
+        test_volt(4 * V, i)
+        test_volt(6 * V, i)
 
-        test_volt(8, i)
-        test_volt(20, i)
-        test_volt(25, i)
+        test_volt(8 * V, i)
```
Comment (deepaksnn) [inline test/pwr/agilent_e36312a.jl:86]: space

Comment (gita135) [reply test/pwr/agilent_e36312a.jl:86]: I made all spaces consistent for units (based on prev written code)

File: test/pwr/agilent_e36312a.jl [hunk 3]
```diff
@@ -1,8 +1,15 @@
 using TcpInstruments
 using Test
 
-pwr = initialize(AgilentE36312A)
-@info "Successfully connected $(pwr.model) at $(get_address(pwr))"
+pwr = initialize(AgilentE36312A, "10.1.30.64")
```
Comment (deepaksnn) [inline test/pwr/agilent_e36312a.jl:4]: Please remove the hardcoded IP address.

Comment (gita135) [reply test/pwr/agilent_e36312a.jl]: Done

**Review by deepaksnn — APPROVED** [review]
👍

**Comment by deepaksnn** [comment]
Looks good. 👍

File: test/impedance_analyzer/agilent4294A.jl [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+end
+
+@testset "Testing get_impedance" begin
+    imped_analyzer_data = get_impedance(imped_analyzer)
+    @info get_impedance imped_analyzer_data
+    @test imped_analyzer_data isa ImpedanceAnalyzerData
+end
+
+@testset "Testing ImpedanceAnalyzerInfo methods" begin
+    @testset "Testing get_volt_dc and get_volt_dc" begin
```
Comment (paul-orchard) [inline test/impedance_analyzer/agilent4294A.jl:47]: small typo, one of them should be get_volt_ac correct?

File: test/pwr/bench_xr.jl [hunk 5]
```diff
@@ -31,53 +37,41 @@ end
...
-    set_current_limit(pwr, "MAX")
-    @info get_current_limit(pwr)
-
-    set_current_limit(pwr, "MIN")
-    @info get_current_limit(pwr)
-
-    test_current(5)
-    test_current(0)
+    test_current(5 * A)
+    test_current(0 * A)
```
Comment (paul-orchard) [inline test/pwr/bench_xr.jl:53]: Is this duplicate to line 43?

File: test/pwr/srsps310.jl [hunk 6]
```diff
@@ -41,21 +48,21 @@ end
 @testset "Voltage" begin
     @info get_voltage(p)
     @info get_voltage_limit(p)
-    set_voltage(p, 40)
-    set_voltage_limit(p, 200)
-    @test get_voltage(p) == 40
+    set_voltage(p, 40 * V)
+    set_voltage_limit(p, 200 * V)
+    @test get_voltage(p) == 40.0 * V
```
Comment (paul-orchard) [inline test/pwr/srsps310.jl:53]: Do we need the extra precision of a floating point for this voltage or should we make it an integer?

**Review by paul-orchard — COMMENTED** [review]
Just a couple of minor comments.

**Comment by gita135** [comment]
> Just a couple of minor comments.

All resolved 👍🏽

**Review by paul-orchard — APPROVED** [review]

**Pull request merged**